### ISSUE TITLE
[CT-3067] [Bug] Docs blocks don't work in model versions

### ISSUE THREAD

**Issue by Gonzalo933** [dbt-labs/dbt-core#8540]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Currently, `.yml` with model definitions do not support using doc blocks inside the fields.

I believe this could be similar to #2913

Commenting or removing the doc block removes the error.

### Expected Behavior

I would expect the doc to be loaded as a string.

### Steps To Reproduce

with this model configuration:

```
version: 2

models:
  - name: working_model
        columns:
          - name: column_id
            description: '{{ doc("column_id_doc") }}'
  - name: error_model
    description: my description
    latest_version: 1
    versions:
      - v: 1
        config:
          alias: my_alias
        columns:
          - name: column_id
            description: '{{ doc("column_id_doc") }}'
      - v: 2
        columns:
          - name: other_id
```

DBT cloud Raises an error when trying to run the tests for the model.

```
Compilation Error
  Could not render {{ doc("column_id_doc") }}: 'doc' is undefined
```

Also, running `dbt docs generate` creates the right documentation for `working_model` and an empty string for the column `column_id` in the model `error_model`

### Environment

```markdown
dbt cloud
```

### Which database adapter are you using with dbt?

snowflake

**Comment by jtcohen6** [comment]
Thanks @Gonzalo933! I was able to reproduce using the example you've provided, and I agree that the intended behavior is to support docs blocks for column-level descriptions at all levels.

Basically, while docs blocks are supported in the `description` field specified for columns at the top level, it isn't supported when specified for `columns` on a specific version:
```yaml
models:
  - name: versioned_model
    description: '{{ doc("any_doc_block") }}'  # this works
    columns:
      - name: id
        description: '{{ doc("any_doc_block") }}'  # this works
    versions:
      - v: 1
        columns:
          - name: id
            description: '{{ doc("any_doc_block") }}'  # this does NOT work
```

**Comment by jtcohen6** [comment]
From estimation:
- Not exactly the same as #8509, but very similar
- Compare to fix in #8709
